File: implementations/fsharp/tests/Skight.AgentPlatform.FSharp.Tests/SystemInfoToolSpecs.fs
namespace Skight.AgentPlatform.FSharp.Tests

open System.IO
open Expecto
open Skight.AgentPlatform.FSharp

module SystemInfoToolSpecs =

    [<Tests>]
    let tests =
        testList "System Info Tool Specs" [

            testAsync "Feature: Fetch Environment and System Metadata" {
                let workspace = Directory.GetCurrentDirectory()
                let! jsonRes = SystemInfoTool.getSystemInfoAsync workspace |> Async.AwaitTask
                Expect.stringContains jsonRes "os" "Response should contain OS metadata"
                Expect.stringContains jsonRes "workspace" "Response should contain workspace path"
            }
        ]

File: implementations/csharp/tests/Skight.AgentPlatform.MSpec.Tests/SystemInfoToolSpecs.cs
using System;
using System.IO;
using Machine.Specifications;
using FluentAssertions;
using Skight.AgentPlatform;

namespace Skight.AgentPlatform.MSpec.Tests
{
    [Subject("System Info Tool Module - Environment Metadata Inspection")]
    public class When_agent_queries_system_info
    {
        Establish context = () =>
        {
            _workspace = Directory.GetCurrentDirectory();
        };

        Because of = () =>
        {
            _jsonResult = SystemInfoTool.GetSystemInfoAsync(_workspace).GetAwaiter().GetResult();
        };

        It should_return_json_containing_os_description = () =>
            _jsonResult.Should().Contain("os");

        It should_contain_machine_and_workspace_information = () =>
            _jsonResult.Should().Contain("workspace");

        static string _workspace;
        static string _jsonResult;
    }
}
